core/unit_test/CMakeLists.txt

```diff
@@ -426,6 +426,7 @@ if(Kokkos_ENABLE_SERIAL)
     UnitTestMainInit.cpp
     ${Serial_SOURCES1}
     serial/TestSerial_Task.cpp
+    serial/TestSerial_MDTeamParallelism.cpp
```

Comment:
Yes, currently commits for each backend placed this `MDTeamParallelism` test only into its section in CMakeList so that it can be compiled and ran without needing implementations from other backends.
Once all of our backend tests are merged, I plan to put in another short PR to convert this into a general test for all backends.